assets/src/edit-story/components/canvas/element.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * External dependencies
+ */
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useLayoutEffect, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { getDefinitionForType } from '../../elements';
+
+const Wrapper = styled.div``;
+
+function Element( {
+	isEditing,
+	addNodeForElement,
+	setClickHandler,
+	handleSelectElement,
+	element: {
+		id,
+		type,
+		...rest
+	},
+} ) {
+	const { Display, Edit } = getDefinitionForType( type );
+	const element = useRef();
+	const props = { ...rest, id };
+
+	useLayoutEffect( () => {
+		addNodeForElement( id, element.current );
+	}, [ id, addNodeForElement ] );
+
+	// Are we editing this element, display this as Edit component.
+	if ( isEditing ) {
+		return (
+			<Wrapper
+				ref={ element }
+			>
+				<Edit { ...props } />
+			</Wrapper>
+		);
+	}
+
+	return (
+		<Wrapper
+			onClick={ ( evt ) => handleSelectElement( id, evt ) }
+			ref={ element }
+		>
+			<Display { ...props } setClickHandler={ setClickHandler } />
```

Comment:
Is there an explanation in one sentence what's the role of the `setClickHandler`?

Reply:
It's a temporary work-around to forward click-events from the selection object to the actual on-page element. In separate work, that @miina's doing, the actual elements will be child elements of the selection and this work-around can be removed then.

Reply:
> Hmm. I'm a bit lost. By "selection object" do you mean a selection DOM element?

I mean the React component `<Selection />`. But this is temporary anyway, this whole mess will be fixed once merged with @miina's efforts elsewere. Better review it then.